fbw-a380x/src/systems/instruments/src/Common/EWDMessageParser.tsx

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+
+const LINE_SPACING = 30;
+const LETTER_WIDTH = 16;
+
+type FormattedFwcTextProps = {
+    x: number,
+    y: number,
+    message: string
+}
+
+const FormattedFwcText: React.FC<FormattedFwcTextProps> = ({ x, y, message }) => {
+    const lines: any[] = [];
+    let spans: any[] = [];
+
+    let color = 'White';
+    let underlined = false;
+    // const flashing = false; TODO
+    let framed = false;
+
+    const decorations: any[] = [];
+
+    let buffer = '';
+    let startCol = 0;
+    let col = 0;
+    for (let i = 0; i < message.length; i++) {
+        const char = message[i];
+        if (char === '\x1b' || char === '\r') {
+            if (buffer !== '') {
+                // close current part
+                spans.push(
+                    <tspan
+                        key={buffer}
+                        className={`${color} EWDWarn`}
+                    >
+                        {buffer}
+                    </tspan>,
+                );
+                buffer = '';
+
+                if (underlined) {
+                    decorations.push(
+                        <path
+                            className={`Underline ${color}Line`}
+                            d={`M ${x + (startCol * LETTER_WIDTH)} ${y + (lines.length * LINE_SPACING + 4)} h ${(col - startCol) * LETTER_WIDTH + 5}`}
+                            strokeLinecap="round"
+                        />,
+                    );
+                }
+
+                if (framed) {
+                    decorations.push(
+                        <path
+                            className={`Underline ${color}Line`}
+                            d={`M ${x + (startCol * LETTER_WIDTH)}
+                            ${y + (lines.length * LINE_SPACING) - 22} h ${(col - startCol) * LETTER_WIDTH + 12} v 27 h ${-((col - startCol) * LETTER_WIDTH + 12)} v -27`}
+                            strokeLinecap="round"
+                        />,
+                    );
+                }
+
+                startCol = col;
+            }
+
+            if (char === '\x1B') {
+                let ctrlBuffer = '';
+                i++;
+                for (; i < message.length; i++) {
```

Comment:
```suggestion
                for (i++; i < message.length; i++) {
```
Also, I think if this index var ``i`` is going to be propagated down like this then it should be so that it's more readable

Reply:
EWDMessageParser wasn't changed by me, and removed via the latest commit (not used for build right now, might be better to refactor to fbw_common).